styx-common/src/main/java/com/spotify/styx/storage/StorageTransaction.java

```diff
@@ -70,7 +70,7 @@ WorkflowId storeWorkflowWithNextNaturalTrigger(Workflow workflow, TriggerInstant
    * @param id Id of the backfill
    * @return Optionally a backfill, if one was found for the given id
    */
-  Optional<Backfill> backfill(String id);
+  Optional<Backfill> backfill(String id) throws IOException;
```

Comment:
Update javadoc as well?

Reply:
We should then add `@throws IOException` to all the other methods in this class as well but that feels like another PR.